src/dimension_slice.c

```diff
@@ -122,13 +122,16 @@ dimension_slice_scan_limit_direction_internal(int indexid, ScanKeyData *scankey,
 		.result_mctx = mctx,
 	};
 
+	if (tuplock)
```

Comment:
Unnecessary `if` clause.

You can assign tuplock directly in the scanctx declaration above.